inference-engine/src/vpu/graph_transformer/src/middleend/passes/allocate_resources.cpp

```diff
@@ -224,6 +224,23 @@ AllocationResult runAllocator(const Model& model, EnableShapeAllocation enableSh
                 allocateShape(tempBuffer);
             }
         }
+
+        // Allocate shape for unused inputs
+        for (const auto& input : model->datas()) {
+            if (input->usage() != DataUsage::Input) {
+                continue;
+            }
+            if (input->numConsumers() != 0 || input->childDataToShapeEdges().size() != 0) {
+                continue;
+            }
```

Comment:
You don't allocate inputs with no consumers and child data -> shape edges - what about similar intermediate and const data objects? Why behavior is different? Why you don't allocate something at all?

Reply:
Discussed it offline. This change is needed to allocate shape for unused inputs. In general, we do this allocation in topological order of stages to make this allocation robust (we can't guarantee that data objects' order in `model->datas()` is the same from run to run which may lead to generating different blobs which is considered as a problem), but we can't reach unused inputs through the topological just because it's not used we make another loop to do so.
But I did this iterating through `model->datas()` which is still not robust, so I've added an additional sorting for unused inputs.